lamindb/core/_compat.py

```diff
@@ -0,0 +1,60 @@
+import importlib.util
+from typing import Any, Callable, TypeVar
+
+T = TypeVar("T")
+
+
+def is_package_installed(package_name: str) -> bool:
+    spec = importlib.util.find_spec(package_name)
+    return spec is not None
+
+
+def with_package(package_name: str, operation: Callable[[Any], T]) -> T:
```

Comment:
Since at least one more person will look at this PR, I make a note here. I know that this is a bit ugly but allows us to write some safer code. Else, we also always have to write lots of statements where we check for available packages, the type of whatever we're looking for, and a PLEASE INSTALL XYZ message. It gets old quick.

Happy to revert if ugly or we can solve it differently. Else, I'll use it also in other places.